programs/ziti-edge-tunnel/ziti-edge-tunnel.c

```diff
@@ -177,6 +177,70 @@ static void on_command_resp(const tunnel_result* result, void *ctx) {
     }
 }
 
+static bool process_tunnel_commands(const tunnel_comand *cmd, command_cb cb, void *ctx) {
+    tunnel_result result = {
+            .success = false,
+            .error = NULL,
+            .data = NULL,
+    };
+    bool cmd_accepted = false;
+    switch (cmd->command) {
+        case TunnelCommand_SetLogLevel: {
+            cmd_accepted = true;
+
+            tunnel_set_log_level tunnel_set_log_level_cmd = {0};
+            if (cmd->data == NULL || parse_tunnel_set_log_level(&tunnel_set_log_level_cmd, cmd->data, strlen(cmd->data)) < 0) {
+                result.error = "invalid command";
+                result.success = false;
+                break;
+            }
+            log_level lvl = log_level_value_of(tunnel_set_log_level_cmd.loglevel);
+            if (lvl != NULL) {
+                if (ziti_log_level() != lvl) {
+                    set_log_level(tunnel_set_log_level_cmd.loglevel);
+                    ziti_log_set_level(lvl);
+                    ZITI_LOG(INFO, "Log level is set to %s", tunnel_set_log_level_cmd.loglevel);
+                } else {
+                    ZITI_LOG(INFO, "Log level is already set to %s", tunnel_set_log_level_cmd.loglevel);
+                }
+                result.success = true;
+            } else {
+                result.error = "invalid loglevel";
+                result.success = false;
+            }
+            break;
+        }
+        case TunnelCommand_UpdateTunIP: {
+            cmd_accepted = true;
+
+            tunnel_tun_ip_v4 tunnel_tun_ip_v4_cmd = {0};
+            if (cmd->data == NULL || parse_tunnel_tun_ip_v4(&tunnel_tun_ip_v4_cmd, cmd->data, strlen(cmd->data)) < 0) {
+                result.error = "invalid command";
+                result.success = false;
+                break;
+            }
+            if (tunnel_tun_ip_v4_cmd.prefixLength < MINTUNIPV4PREFIXLENGTH || tunnel_tun_ip_v4_cmd.prefixLength > MAXTUNIPV4PREFIXLENGTH) {
+                char error_msg[50];
+                snprintf(error_msg, sizeof(error_msg), "prefix length should be between %d and %d", MINTUNIPV4PREFIXLENGTH, MAXTUNIPV4PREFIXLENGTH);
+                result.error = error_msg;
```

Comment:
The whole tunnel_result is sent as json to the ipc queue as response

Reply:
even so, if anyone tried to access `result.error` after this code block it would crash or lead to unexpected results.

Can you point me to the place where result.error is actually used?

Reply:
I changed this error statement to 
result.error = "prefix length should be between 10 and 18";

It is used in the callback function - on_command_resp. Here we convert the tunnel_result to json and send it back to the IPC queue
static void on_command_resp(const tunnel_result* result, void *ctx) {
    size_t json_len;
    char *json = tunnel_result_to_json(result, MODEL_JSON_COMPACT, &json_len);